packages/flutter_tools/test/general.shard/runner/flutter_command_runner_test.dart

```diff
@@ -86,6 +88,24 @@ void main() {
         Platform: () => platform,
       }, initializeFlutterRoot: false);
 
+      testUsingContext('does not check that Flutter installation is up-to-date with CI=true in environment', () async {
```

Comment:
Add tests for your `explicitVersionCheckPassed ` logic: adding `--version-check` explicitly overrides `--machine` and the bot detection.

Reply:
Good idea! - Done